relay-server/src/actors/events.rs

```diff
@@ -855,19 +941,32 @@ impl Handler<QueueEnvelope> for EventManager {
             return Err(QueueEnvelopeError::TooManyEvents);
         }
 
-        self.current_active_events += 1;
-
         let event_id = message.envelope.event_id();
 
-        // Actual event handling is performed asynchronously in a separate future. The lifetime of
-        // that future will be tied to the EventManager's context. This allows to keep the Project
-        // actor alive even if it is cleaned up in the ProjectManager.
+        // Split the envelope into event-related items and other items. This allows to fast-track:
+        //  1. Envelopes with only session items. They only require rate limiting.
+        //  2. Event envelope processing can bail out if the event is filtered or rate limited,
+        //     since all items depend on this event.
+        if let Some(event_envelope) = message.envelope.split_by(Item::requires_event) {
```

Comment:
This split here is crucial since this is the reason we can still bail out in so many cases (see the code comment above).